src/coreclr/interop/comwrappers.cpp

```diff
@@ -338,6 +338,26 @@ namespace
     static_assert(sizeof(ManagedObjectWrapper_IReferenceTrackerTargetImpl) == (7 * sizeof(void*)), "Unexpected vtable size");
 }
 
+namespace
+{
+    // 5c13e51c-4f32-4726-a3fd-f3edd63da3a0
+    const GUID IID_TaggedImpl = { 0x5c13e51c, 0x4f32, 0x4726, { 0xa3, 0xfd, 0xf3, 0xed, 0xd6, 0x3d, 0xa3, 0xa0 } };
```

Comment:
Do we need to disambiguate wrappers from different runtimes? E.g. consider both standard coreclr and nativeaot compiled binary running in the same process; or multiple nativeaot compiled binaries running in the same process.

Reply:
> Do we need to disambiguate wrappers from different runtimes?

That is a great question. Historically no, but with NativeAOT that might be something to consider. I think having a way to disambiguate is a good idea because people just won't know and from the CoreCLR perspective a NativeAOT impl is really native and a black box.

Reply:
From the NativeAOT perspective this does become a version boundary though. Consider the case where we have NativeAOT v1 and NativeAOT v2. If the CCW layout remain the same, then either runtime can do unwrapping. However, if v2 changes the layout then it shouldn't perform the action on a v1 layout. That is a tough issue now that I think about it. Should we reconsider this?

Reply:
The way you to solve this is to make TaggedImpl return a pointer that unique represents the runtime instance. It can be a method on the interface that returns the pointer. (Alternatively, you can use address of the `QueryInterface` method in the interface vtable if would like cheat a bit.)

Reply:
I can do the pointer return trick. Thanks. Any suggestions on what pointer value to use in NativeAOT? `MethodTable` for `object` or something like that? I've no idea how to get at that but that seems unique enough.

Reply:
It can be the same pointer in all runtimes. What is the pointer you intend to use for CoreCLR?

Reply:
I think the address of the `MethodTable` for `object`. It is easy/fast to acquire in CoreCLR.

Reply:
Nope. I stand corrected. The `ComWrappers` native impl is in a static lib and doesn't have access to the `MethodTable` for `object`. I'm going another direction.